qiskit_experiments/database_service/db_fitval.py

```diff
@@ -34,3 +37,89 @@ def __str__(self):
         if self.unit is not None:
             out += f" {str(self.unit)}"
         return out
+
+    def __new__(cls, *args, **kwargs):
+        # Note that FitVal can be instantiated from the json loader thus
+        # DeprecationWarning is not captured by default execution setting.
+        # It can be only seen if the code is executed from __main__.
+        warnings.warn(
+            "The FitVal class is deprecated as of Qiskit Experiments 0.3 and will"
+            " be removed in a future version. Re-saving loaded experiment data or "
+            " analysis results will convert FitVals to ExperimentVariable.",
+            FutureWarning,
+        )
+        if len(args) > 0:
+            nominal_value = args[0]
+        else:
+            nominal_value = kwargs.get("value")
+        if len(args) > 1:
+            std_dev = args[1]
+        else:
+            std_dev = kwargs.get("stderr")
+        if len(args) > 2:
+            tag = args[2]
+        else:
+            tag = kwargs.get("unit")
+
+        try:
+            # This might be fitting parameter entries
+            # This has been storing list of parameters and stdevs as a single FitVal
+            # Stdevs are no longer necessary since it can be computed with covariance matrix.
+            nominal_values = iter(nominal_value)
+            return list(nominal_values)
+        except TypeError:
+            return uncertainties.core.Variable(
+                value=nominal_value,
+                std_dev=std_dev,
+                tag=tag,
+            )
+
+
+# Monkey patch uncertainties UFloat class so that it behaves like
+# FitVal with deprecation warnings when used as a replacement for
+# for analysis result value types
+
+
+def value(self):
+    """DEPRECATED"""
+    warnings.warn(
+        "The FitVal class has been depreacted and replaced with UFloat "
+        "objects, use .nominal_value or .n to access the equivalent of "
+        "the FitVal.value property",
+        DeprecationWarning,
+        stacklevel=2,
+    )
+    # deprecation warning
+    return self.nominal_value
+
+
+def stderr(self):
+    """DEPRECATED"""
+    warnings.warn(
+        "The FitVal class has been depreacted and replaced with UFloat "
+        "objects, use .std_dev or .s to access the equivalent of the "
+        "FitVal.stderr property.",
+        DeprecationWarning,
+        stacklevel=2,
+    )
+    # deprecation warning
+    return self.std_dev
+
+
+def unit(self):
+    """DEPRECATED"""
+    warnings.warn(
+        "The FitVal class has been depreacted and replaced with UFloat "
+        "objects which do not contain units. This will return the .tag "
+        "property which may be equivalent to the FitVal.unit property "
+        "if constructed from a loaded FitVal.",
+        DeprecationWarning,
+        stacklevel=2,
+    )
+    return self.tag
```

Comment:
No because `FitVal` is always typecasted into `Variable` which has `.tag`. We don't encounter `ScalarAffineFunc` until we operate two values but this was not expected behavior since `FitVal`s are not directly mathematically operated.

Reply:
This makes me think we should monkey patch `uncertainties.core.Variable` rather than `AffineScalarFunc`.

Reply:
Ok, I guess this is fine then